pkg/fleet/internal/cdn/cdn.go

```diff
@@ -36,38 +56,125 @@ func New(env *env.Env) *CDN {
 }
 
 // Get gets the configuration from the CDN.
-func (c *CDN) Get(ctx context.Context) (_ Config, err error) {
+func (c *CDN) Get(ctx context.Context) (_ *Config, err error) {
 	span, ctx := tracer.StartSpanFromContext(ctx, "cdn.Get")
 	defer func() { span.Finish(tracer.WithError(err)) }()
-	return getFakeCDNConfig(ctx)
+	configLayers, err := c.getOrderedLayers(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return newConfig(configLayers...)
 }
 
-// HACK (arthur): this is a temporary function that returns a fake CDN config to unblock development.
-func getFakeCDNConfig(_ context.Context) (Config, error) {
-	baseLayer := layer{
-		ID: "base",
-		Content: map[interface{}]interface{}{
-			"extra_tags": []string{"layer:base"},
-		},
+// getOrderedLayers calls the Remote Config service to get the ordered layers.
+// Today it doesn't use the CDN, but it should in the future
+func (c *CDN) getOrderedLayers(ctx context.Context) ([]*layer, error) {
+	// HACK(baptiste): Create a dedicated one-shot RC service just for the configuration
+	// We should use the CDN instead
+	config := pkgconfigsetup.Datadog()
+	config.Set("run_path", "/opt/datadog-packages/datadog-installer/stable/run", model.SourceAgentRuntime)
+
+	detectenv.DetectFeatures(config)
+	hostname, err := pkghostname.Get(ctx)
+	if err != nil {
+		return nil, err
 	}
-	overrideLayer := layer{
-		ID: "override",
-		Content: map[interface{}]interface{}{
-			"extra_tags": []string{"layer:override"},
-		},
+
+	options := []remoteconfig.Option{
+		remoteconfig.WithAPIKey(c.env.APIKey),
+		remoteconfig.WithConfigRootOverride(c.env.Site, ""),
+		remoteconfig.WithDirectorRootOverride(c.env.Site, ""),
 	}
-	config, err := newConfig(baseLayer, overrideLayer)
+	service, err := remoteconfig.NewService(
+		config,
+		"Datadog Installer",
+		fmt.Sprintf("https://config.%s", c.env.Site),
+		hostname,
+		getHostTags(ctx, config),
+		&rctelemetryreporterimpl.DdRcTelemetryReporter{}, // No telemetry for this client
+		"9.9.9",
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏Let's change it now, or we'll forget to update it